pkg/resource/deploy/step_generator.go

```diff
@@ -585,8 +596,13 @@ func (sg *stepGenerator) generateStepsFromDiff(
 			//
 			// Note that if we're performing a targeted replace, we already have the correct inputs.
 			if prov != nil && !sg.isTargetedReplace(urn) {
+				// Increment the sequence number (if it's known) before calling check so we get a new autoname
+				if new.SequenceNumber != 0 {
```

Comment:
This is the main place where we're incrementing it seems. Right?

Why is it under `prov != nil && !sg.isTargetedReplace(urn)` condition? Looks like otherwise we don't even call Check. In other words, if we *are* doing targeted replace, will sequence auto-increment?

Reply:
So I think the code higher in the file (~350) should handle this case now. If we're recreating or doing a targeted replace we need to increment.